testing/tests/consensus_task.rs

```diff
@@ -0,0 +1,317 @@
+    TYPES: NodeType<
+        ConsensusType = SequencingConsensus,
+        ElectionConfigType = StaticElectionConfig,
+        SignatureKey = JfPubKey<BLSSignatureScheme<ark_bls12_381::Parameters>>,
+    >,
+    I: TestableNodeImplementation<
+        TYPES::ConsensusType,
+        TYPES,
+        Leaf = SequencingLeaf<TYPES>,
+        ConsensusMessage = SequencingMessage<TYPES, I>,
+        Storage = MemoryStorage<SequencingTestTypes, SequencingLeaf<SequencingTestTypes>>,
+    >,
+>(
+    task_runner: TaskRunner,
+    event_stream: ChannelStream<SequencingHotShotEvent<TYPES, I>>,
+) -> TaskRunner
+where
+    I::Exchanges: SequencingExchangesType<
+        TYPES,
+        Message<TYPES, I>,
+        Networks = (StaticQuroumComm, StaticDAComm),
+    >,
+    SequencingQuorumEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Proposal = QuorumProposal<TYPES, SequencingLeaf<TYPES>>,
+        Certificate = QuorumCertificate<TYPES, SequencingLeaf<TYPES>>,
+        Commitment = SequencingLeaf<TYPES>,
+    >,
+    CommitteeEx<TYPES, I>: ConsensusExchange<
+        TYPES,
+        Message<TYPES, I>,
+        Certificate = DACertificate<TYPES>,
+        Commitment = TYPES::BlockType,
+    >,
+{
+    let builder = TestBuilder::default_multiple_rounds();
+
+    let launcher = builder.build::<SequencingTestTypes, SequencingMemoryImpl>();
+    let node_id = 1;
+    let network_generator = Arc::new((launcher.generator.network_generator)(node_id));
+    let quorum_network = (launcher.generator.quorum_network)(network_generator.clone());
+    let committee_network = (launcher.generator.committee_network)(network_generator);
+    let storage = (launcher.generator.storage)(node_id);
+    let config = launcher.generator.config.clone();
+    let initializer =
+        HotShotInitializer::<TYPES, I::Leaf>::from_genesis(I::block_genesis()).unwrap();
+
+    let node_id = 1;
+
+    let known_nodes = config.known_nodes.clone();
+    let private_key = I::generate_test_key(node_id);
+    let public_key = TYPES::SignatureKey::from_private(&private_key);
+    let ek =
+        jf_primitives::aead::KeyPair::generate(&mut rand_chacha::ChaChaRng::from_seed([0u8; 32]));
+    let election_config = config.election_config.clone().unwrap_or_else(|| {
+        <QuorumEx<TYPES,I> as ConsensusExchange<
+                TYPES,
+                Message<TYPES, I>,
+            >>::Membership::default_election_config(config.total_nodes.get() as u64)
+    });
+    let exchanges = I::Exchanges::create(
+        known_nodes.clone(),
+        election_config.clone(),
+        (quorum_network, committee_network),
+        public_key.clone(),
+        private_key.clone(),
+        ek.clone(),
+    );
+    let handle = SystemContext::init(
+        public_key,
+        private_key,
+        node_id,
+        config,
+        storage,
+        exchanges,
+        initializer,
+        NoMetrics::boxed(),
+    )
+    .await
+    .expect("Could not init hotshot");
```

Comment:
Is this for the unit tests? Or is this the way consensus is initialized? In the latter case, we should discuss this. I'm already initializing the systemcontexts in the testing framework.

Reply:
Yeah this is just for unit tests, again kind of a hack to get the exchanges to create the consensus task.